features/22-email-verification/api.md

```diff
@@ -1,73 +1,133 @@
+* User may request the verification link or code to be resent.
+* App may toggle the verification status of any user through cloud code or SDKs.
 
 ## Requirements
 
-* Server will have configuration options for:
-	* enable/disable email verification (boolean)
-	* enable/disable SMS verification (boolean)
-	* set verification as required/optional (boolean)
-	* smtp server settings
-	* SMS gateway settings
-	* email template
-	* SMS template
-* Server will add API for sending/resending verification email
-	* when called by an admin or with master key, the API can send verification email to any unverified user, this is designed for cloud code use
-	* when called by non-admin unverified user, this API resends the verification email, this is designed for app use
-* Server will add API for checking email verification code
-* Server will add a request preprocessor to check if the user is verified and return an error (if verification is required).
-* Signup/login API to return verification status of a user
-* Signup API to send verification email if verification is enabled.
-* JS, iOS and Android SDK to add API for resending verification email and checking verification code.
+### User Verified Flag
+
+Each user will have a flag that indicates whether the user is already
+verified. This flag is internal to Skygear for authentication and access
+control purpose.
+
+The Server will implements new actions to support toggling the user verified
+flag. SDKs will add support for calling the new actions to toggle the user
+verified flag.
+
+The flag can also be configured to automatically updated when user data changes.
+
+The flag is readable for the current user.
+
+### Supported Verifiable User Data
+
+The server will support verifying these user data:
+
+* Email address: By sending a email to the email address through SMTP.
+* Phone number: By sending a SMS to the phone number through a SMS gateway. Only
+  supporting Twilio and Nexmo for the time being.
+
+### Storing User Data and Data Verified Flag
+
+User data (i.e. email address and phone number) that requires verification
+must be saved in user profile (i.e. in Record DB under `user` record type).
+The field name for the user data is configurable.
+
+Data verified flag (i.e. whether the email address or phone number is 
+verified) will be saved in the user profile, alongside the user data
+to be verified. This allows the app to use Record DB API to query the data
+verified flag in order to display the information to user.
 
 ### Criteria for Verifying a User
 
-The developer is able to specify which information is required to be verified
+The developer is able to specify which user data is required to be verified
 in order for a user to become verified. For example, an app that requests for
 email and phone may require either or both email and phone is verified before
 the user become verified.
 
-### Verification Code Persistence and Consumption
+### Configuration
```

Comment:
I think similar to forgotPassword, we will need to have configurations for the HTML or URL of the verification success/failure page?